### PR TITLE
Build PDB file for Windows

### PR DESCRIPTION
**Pull request by zyn0217**
This could help with dumping human-readable backtrace symbols with line no.

See also #1641.

### CODE REVIEW

**Comment by zyn0217** [comment]
Request for review :)

@sam-mccall @kadircet @hokein 

CC @HighCommander4

File: .github/ISSUE_TEMPLATE/bug_report.md
Comment (sam-mccall) [inline .github/ISSUE_TEMPLATE/bug_report.md:20]: rather than requiring this extra step, it'd be nice just to include the pdb file in the right location in the windows clangd.zip

Do you think it is too large in practice?
I like having the binaries be reasonably compact

(Ideally I think we'd ship by default a tiny debuginfo that just had just enough to symbolize stacktraces - this is trivially small as `-gmlt` on other platforms shows - but don't want to ask you to dig into that if it's not really obvious)

**Review by sam-mccall — APPROVED** [review]
Sorry about the delay here, I've been missing stuff coming through github :-(

This looks OK to me, I'd probably prefer to just ship the PDB by default unless it's big (say, >50% of clangd.exe) but will leave that up to you.

**Comment by zyn0217** [comment]
Sorry for my procrastination; I didn't get my hands on my Windows device for a while ;)

I did a rough test with a raw PDB file of around 760 MiB locally: (Compression method - Archive size - Running time)

- bzip2: 102 MiB; 34 s
- deflate: 138 MiB; 77 s
- deflate64: 128 MiB; 90 s
- lzma: 74 MiB; 40.7 s
- ppmd: 92 MiB; 36 s

All of these methods are running with default arguments that 7zip suggests. Of the result, LZMA wins with a fair running time.

@sam-mccall WDYT? Would you mind merging this PR so that we could roll the clangd 17 release as well?

File: .github/workflows/autobuild.yaml
Comment (zyn0217) [inline .github/workflows/autobuild.yaml:156]: Note I'm turning the parallel link jobs for Windows builder to 1 since there's only 7 GiB available for [Github's free action plan](https://docs.github.com/en/actions/using-github-hosted-runners/about-github-hosted-runners/about-github-hosted-runners#supported-runners-and-hardware-resources). Running these jobs together would somehow hit this bottleneck and results in [OOM](https://github.com/zyn0217/clangd/actions/runs/6439605610/job/17487472454).

And here's the [action link](https://github.com/zyn0217/clangd/actions/runs/6440816610) to my successfully built [artifacts](https://github.com/zyn0217/clangd/releases/tag/17.0.2). (Windows only)

**Comment by zyn0217** [comment]
Gently ping @sam-mccall

**Comment by mika-fischer** [comment]
Anybody able to move this forward?

**Comment by zyn0217** [comment]
@HighCommander4 WDYT? Can we merge this PR (I don't have the commit access to this repository) so that we can roll out some PDB files in the upcoming 19 release?

File: .github/ISSUE_TEMPLATE/bug_report.md
Comment (HighCommander4) [inline .github/ISSUE_TEMPLATE/bug_report.md:20]: It would be great to add this information to https://clangd.llvm.org/troubleshooting#getting-stacktraces-from-crashes as well (the repo is at https://github.com/llvm/clangd-www/blob/main/troubleshooting.md).

**Review by HighCommander4 — COMMENTED** [review]
> @HighCommander4 WDYT? Can we merge this PR (I don't have the commit access to this repository) so that we can roll out some PDB files in the upcoming 19 release?

I've looked over the patch and discussion, and it looks to me like this should be good to merge. We can explore putting the PDB file into the same package as the binary in a separate change, if there is a desire for that.

**Comment by HighCommander4** [comment]
Note, the patch needs a rebase before it can be merged.

**Comment by HighCommander4** [comment]
> Note, the patch needs a rebase before it can be merged.

Actually, the merge conflict was simple to resolve so I did that myself.

File: .github/ISSUE_TEMPLATE/bug_report.md
Comment (zyn0217) [reply .github/ISSUE_TEMPLATE/bug_report.md:20]: Thanks, I will put up a PR then.

**Pull request merged**
